src/explorer/WebAppProvider.ts

```diff
@@ -31,22 +31,18 @@ export class WebAppProvider implements IChildProvider {
 
         this._nextLink = webAppCollection.nextLink;
 
-        return await Promise.all(webAppCollection
-            .filter((s: Site) => {
+        const treeItems: IAzureTreeItem[] = [];
+        await Promise.all(webAppCollection
+            .map(async (s: Site) => {
                 try {
-                    const siteClient = new SiteClient(s, node);
-                    return siteClient !== undefined;
-                    // tslint:disable-next-line:no-unsafe-any
-                } catch (err) {
-                    return false;
+                    const siteClient: SiteClient = new SiteClient(s, node);
+                    const appServicePlan: AppServicePlan = await siteClient.getAppServicePlan();
+                    treeItems.push(new WebAppTreeItem(siteClient, appServicePlan));
+                } catch {
+                    treeItems.push(new InvalidWebAppTreeItem(`${s.name} (invalid)`));
```

Comment:
I like 'invalid'. However, rather than doing the formatting yourself, please specify `s.name` as the `label` property and `Invalid` as the `description` property

That will ensure its displayed in the same format as our other description properties (which as of today is the same format that you did, but could change in the future)

Reply:
Description property added.